library/src/main/java/com/pokegoapi/api/pokemon/PokemonDetails.java

```diff
@@ -305,7 +305,10 @@ public int getMaxCpFullEvolveAndPowerupForPlayer() throws LoginFailedException,
 	 */
 	private int getMaxCpFullEvolveAndPowerup(int playerLevel) {
 		PokemonIdOuterClass.PokemonId highestUpgradedFamily;
-		if (getPokemonId() == EEVEE) {
+		if (asList(VAPOREON, JOLTEON, FLAREON).contains(getPokemonId())) {
+			highestUpgradedFamily = getPokemonId();
+		}
+		else if (getPokemonId() == EEVEE) {
```

Comment:
Are you sure checkstyle runs without errors. Last time with this it prints an error